spec/domain/mailing_lists/subscribers_spec.rb

```diff
@@ -70,23 +82,64 @@
             bottom_member.tag_list = %w[foo]
             bottom_member.save!
 
-            expect(list.people).to eq [bottom_member]
+            expect(subject).to eq [bottom_member]
           end
 
-          it "includes person member if it lacks excluding tag" do
+          it "includes person if it lacks excluding tag" do
             sub = create_subscription(bottom_layer_one, false, Group::BottomLayer::Member.sti_name)
             sub.subscription_tags = subscription_tags(%w[foo], excluded: true)
             sub.save!
-            expect(list.people).to eq [bottom_member]
+            expect(subject).to eq [bottom_member]
           end
 
-          it "excludes person member if matches excluding tag" do
+          it "excludes person if it matches excluding tag" do
             sub = create_subscription(bottom_layer_one, false, Group::BottomLayer::Member.sti_name)
             sub.subscription_tags = subscription_tags(%w[foo], excluded: true)
             sub.save!
             bottom_member.tag_list = %w[foo]
             bottom_member.save!
-            expect(list.people).to be_empty
+            expect(subject).to be_empty
+          end
+
+          it "excludes person if it matches any excluding tag" do
+            sub = create_subscription(bottom_layer_one, false, Group::BottomLayer::Member.sti_name)
+            sub.subscription_tags = subscription_tags(%w[foo baz], excluded: true)
+            sub.save!
+            bottom_member.tag_list = %w[foo bar]
+            bottom_member.save!
+            expect(subject).to be_empty
+          end
+
+          it "includes person if first role is excluded by tag but second role is not" do
+            sub = create_subscription(bottom_layer_one, false, Group::BottomLayer::Member.sti_name)
+            sub.subscription_tags = subscription_tags(%w[foo baz], excluded: true)
+            sub.save!
+
+            create_subscription(bottom_layer_one, false, Group::BottomLayer::Leader.sti_name)
+
+            Fabricate(Group::BottomLayer::Leader.sti_name, group: bottom_layer_one, person: bottom_member)
+
+            bottom_member.tag_list = %w[foo bar]
+            bottom_member.save!
+
+            expect(subject).to eq [bottom_member]
+          end
```

Comment:
Beim manuellen/script Testen mit einem (etwas älteren) Prod Datenstand ist mir aufgefallen dass dies im `master` aktuell nicht so funktioniert.

Wird eine Person durch ein Tag in irgend einer Group Subscription exkludiert taucht sie in der Liste nicht auf. Egal ob sie via andere Group Subscription wieder inkludiert wird.
So wie es aber jetzt gelöst ist finde ich es aber definitiv auch nicht falsch, ist halt Auslegungssache der Benutzersicht.

Ich kann sonst noch versuchen @ThomasEllenberger zu erreichen um seine Meinung einzuholen. Bei der Liste bei der es aber mir auffiel war der Unterschied zwei Personen in der Geschäftsstelle. Sonst sah alles identisch aus.